plugin/src/main/java/org/jenkinsci/plugins/structs/describable/DescribableModel.java

```diff
@@ -284,13 +296,24 @@ public T instantiate(Map<String,?> arguments) throws Exception {
             arguments = Collections.singletonMap(rp.getName(),arguments.get(ANONYMOUS_KEY));
         }
 
+        Set<String> erroneous =  new TreeSet<>(arguments.keySet());
+        erroneous.removeAll(parameters.keySet());
+        if (erroneous.size() > 0) {
+            String msg = "Unknown parameter(s) found for class type '" + this.type.getName() + "': " + String.join(",", erroneous);
+            if (Main.isUnitTest) {
+                throw new IllegalArgumentException(msg);
+            } else {
+                listener.getLogger().format("WARNING: %s", msg);
+            }
+        }
+
         try {
-            Object[] args = buildArguments(arguments, constructor.getGenericParameterTypes(), constructorParamNames, true);
+            Object[] args = buildArguments(arguments, constructor.getGenericParameterTypes(), constructorParamNames, true, listener);
```

Comment:
It looks like you aren't actually using the listener anywhere inside of `buildArguments` or the methods it calls (likewise for `injectSetters` below). I would remove the new parameter, and if someone in the future has a need for it, they can thread it through to where it is needed. This class is `final` anyway, so it's not like a subclass might want to use the listener in an override or something.

Reply:
I'm actually using it in both functions for the `coerce` method calls in `line 391` and `line 411`

Reply:
Ok, makes sense, nevermind then!